src/sys/stat.rs

```diff
@@ -7,14 +6,357 @@ pub use libc::c_uint;
+    }
+
+    #[cfg(all(
+        target_os = "android",
+        any(target_arch = "x86_64", target_arch = "aarch64")
+    ))]
+    /// Returns the number of blocks allocated
+    pub fn blocks(&self) -> libc::c_long {
+        self.0.st_blocks
+    }
+
+    #[cfg(all(
+        target_os = "android",
+        any(target_arch = "arm", target_arch = "x86", target_arch = "armv7")
+    ))]
+    /// Returns the number of blocks allocated
+    pub fn blocks(&self) -> libc::c_ulonglong {
+        self.0.st_blocks
+    }
+
+    #[cfg(not(all(
+        target_os = "android",
+        any(
+            target_arch = "aarch64",
+            target_arch = "x86_64",
+            target_arch = "arm",
+            target_arch = "armv7",
+            target_arch = "x86"
+        )
+    )))]
+    /// Returns the number of blocks allocated
+    pub fn blocks(&self) -> blkcnt_t {
+        self.0.st_blocks
+    }
+
+    #[cfg(not(target_os = "netbsd"))]
+    /// Returns the time of last access
+    pub fn atime(&self) -> TimeSpec {
+        TimeSpec::from_timespec(timespec {
+            tv_sec: self.0.st_atime,
+            tv_nsec: self.0.st_atime_nsec,
+        })
+    }
+
+    #[cfg(target_os = "netbsd")]
+    /// Returns the time of last access
+    pub fn atime(&self) -> TimeSpec {
+        TimeSpec::from_timespec(timespec {
+            tv_sec: self.0.st_atime,
+            tv_nsec: self.0.st_atimensec,
+        })
+    }
+
+    #[cfg(not(target_os = "netbsd"))]
+    /// Returns the time of last modification
+    pub fn mtime(&self) -> TimeSpec {
+        TimeSpec::from_timespec(timespec {
+            tv_sec: self.0.st_mtime,
+            tv_nsec: self.0.st_mtime_nsec,
+        })
+    }
+
+    #[cfg(target_os = "netbsd")]
+    /// Returns the time of last modification
+    pub fn mtime(&self) -> TimeSpec {
+        TimeSpec::from_timespec(timespec {
+            tv_sec: self.0.st_mtime,
+            tv_nsec: self.0.st_mtimensec,
+        })
+    }
+
+    #[cfg(not(target_os = "netbsd"))]
+    /// Returns the time of last status change
+    pub fn ctime(&self) -> TimeSpec {
+        TimeSpec::from_timespec(timespec {
+            tv_sec: self.0.st_ctime,
+            tv_nsec: self.0.st_ctime_nsec,
+        })
+    }
+
+    #[cfg(target_os = "netbsd")]
+    /// Returns the time of last status change
+    pub fn ctime(&self) -> TimeSpec {
+        TimeSpec::from_timespec(timespec {
+            tv_sec: self.0.st_ctime,
+            tv_nsec: self.0.st_ctimensec,
+        })
+    }
+
+    #[cfg(any(target_os = "openbsd", target_os = "macos", target_os = "ios"))]
+    /// Return the file flags
+    pub fn flags(&self) -> FileFlag {
+        FileFlag::from_bits(self.0.st_flags).unwrap()
+    }
+
+    #[cfg(any(
+        target_os = "freebsd",
+        target_os = "dragonfly",
+        target_os = "netbsd",
+    ))]
+    /// Return the file flags
+    pub fn flags(&self) -> FileFlag {
+        FileFlag::from_bits(self.0.st_flags.into()).unwrap()
```

Comment:
On `freebsd/dragonfly/netbsd`, `st_flags` is defined as `u32`, but `type_of_file_flag`, a generic type for `FileFlag`, has type `c_ulong` 
```rust
#[cfg(any(target_os = "macos", target_os = "ios", target_os = "openbsd"))]
pub type type_of_file_flag = c_uint;
#[cfg(any(
    target_os = "netbsd",
    target_os = "freebsd",
    target_os = "dragonfly"
))]
pub type type_of_file_flag = c_ulong;

#[cfg(any(
    target_os = "openbsd",
    target_os = "netbsd",
    target_os = "freebsd",
    target_os = "dragonfly",
    target_os = "macos",
    target_os = "ios"
))]
libc_bitflags! {
    /// File flags.
    #[cfg_attr(docsrs, doc(cfg(all())))]
    pub struct FileFlag: type_of_file_flag {

```

And the constants of `FileFlag` are defined as `c_ulong`, as well as the arguments of `chflags/fchflags/lchflags/chflagsat`
```rust
pub const UF_SPARSE: c_ulong = 0x00000100;
```
```rust
pub unsafe extern "C" fn chflags( path: *const c_char, flags: c_ulong) -> c_int
pub unsafe extern "C" fn fchflags(fd: c_int, flags: c_ulong) -> c_int
pub unsafe extern "C" fn lchflags( path: *const c_char, flags: c_ulong) -> c_int
pub unsafe extern "C" fn chflagsat(fd: c_int, path: *const c_char, flags: c_ulong, atflag: c_int) -> c_int
```

So I use `.into()` here to convert `u32` to `u64`